### PR TITLE
METRON-1654: findOne request after an alert patch returns with the original state of the alert item

### PR DESCRIPTION
**Pull request by merrimanr**
## Contributor Comments
With the addition of https://github.com/apache/metron/pull/1190, the Alerts UI can now update state using the object returned from a REST update call.  This PR updates the Alerts UI to do that.

### Testing
This has been tested in full dev.  

1. Spin up full dev and navigate to the Alerts UI
2. Open an alert and change the status from "NEW" to "OPEN".  After closing the detail panel the status for that alert should show as "OPEN" in the list view.
3. Create a metaalert and search for the metaalert so that is shows up in the list view.  Add an alert to that metaalert by checking the checkbox next to an alert and using the "Add to Alert" option in the "Actions" dropdown.  The count next the the metaalert in the list view should increment by 1.
4. Remove an alert from the metaalert.  The count next to the metaalert should decrement by 1 and the alert should be removed immediately.
5. Other update functions in the UI should be tested to ensure there are no regressions:  adding/deleting a comment, changing a metaalert name, etc.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [ ] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

**Comment by nickwallen** [comment]
@merrimanr When you get a chance, we're conflicted.

File: metron-interface/metron-alerts/src/app/alerts/alert-details/alert-details.component.ts [hunk 1]
```diff
@@ -252,22 +234,15 @@ export class AlertDetailsComponent implements OnInit {
...
-        let deletedCommentWrapper = this.alertCommentsWrapper.splice(index, 1)[0];
         let commentRequest = new CommentAddRemoveRequest();
         commentRequest.guid = this.alertSource.guid;
         commentRequest.comment = this.alertCommentsWrapper[index].alertComment.comment;
         commentRequest.username = this.alertCommentsWrapper[index].alertComment.username;
         commentRequest.timestamp = this.alertCommentsWrapper[index].alertComment.timestamp;
         commentRequest.sensorType = this.alertSourceType;
-        this.updateService.removeComment(commentRequest).subscribe(
-            () => {
-              this.alertCommentsWrapper.map(alertsWrapper => alertsWrapper.alertComment)
```
[the remark this answers is not in the document]
Comment (merrimanr) [reply metron-interface/metron-alerts/src/app/alerts/alert-details/alert-details.component.ts]: It's the opposite now.  Before, as you said, we were doing nothing on success and reverting optimistic changes on error.  Now we update on success and do nothing on error.  So no need to handle the error case unless there is something else we want to do with it.

File: metron-interface/metron-alerts/src/app/alerts/alerts-list/alerts-list.component.ts [hunk 2]
```diff
@@ -456,20 +460,17 @@ export class AlertsListComponent implements OnInit, OnDestroy {
...
 
-    this.searchService.getAlert(sensorType, guid).subscribe(alertSource => {
-      this.alerts.filter(alert => alert.source.guid === guid)
-      .map(alert => alert.source = alertSource);
+  removeAlert(alertSource: AlertSource) {
+    let alertIndex = -1;
+    this.alerts.forEach((alert, index) => {
+      alertIndex = (alert.source.guid === alertSource.guid) ? index : alertIndex;
     });
+    this.alerts.splice(alertIndex, 1);
```
[the remark this answers is not in the document]
Comment (merrimanr) [reply metron-interface/metron-alerts/src/app/alerts/alerts-list/alerts-list.component.ts]: Wow you're right.  Filter is much better.

File: metron-interface/metron-alerts/src/app/alerts/alert-details/alert-details.component.ts [hunk 3]
```diff
@@ -252,22 +234,15 @@ export class AlertDetailsComponent implements OnInit {
 
     const confirmedSubscription = this.dialogService.launchDialog(commentText).subscribe(action => {
       if (action === ConfirmationType.Confirmed) {
-        let deletedCommentWrapper = this.alertCommentsWrapper.splice(index, 1)[0];
```
[the remark this answers is not in the document]
Comment (merrimanr) [reply metron-interface/metron-alerts/src/app/alerts/alert-details/alert-details.component.ts]: Ah breathe it in

File: metron-interface/metron-alerts/src/app/alerts/alerts-list/alerts-list.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (merrimanr) [reply metron-interface/metron-alerts/src/app/alerts/alerts-list/alerts-list.component.ts]: Done

File: metron-interface/metron-alerts/src/app/alerts/alerts-list/tree-view/tree-view.component.ts [hunk 4]
```diff
@@ -436,16 +437,13 @@ export class TreeViewComponent extends TableViewComponent implements OnInit, OnC
...
-          group.response.results.filter(alert => alert.source.guid === patchRequest.guid)
-          .map(alert => alert.source = alertSource);
-        }
-      });
+  updateAlert(alertSource: AlertSource) {
+    Object.keys(this.treeGroupSubscriptionMap).forEach(key => {
+      let group = this.treeGroupSubscriptionMap[key].group;
+      if (group.response && group.response.results && group.response.results.length > 0) {
+        group.response.results.filter(alert => alert.source.guid === alertSource.guid)
+        .map(alert => alert.source = alertSource);
```
[the remark this answers is not in the document]
Comment (merrimanr) [reply metron-interface/metron-alerts/src/app/alerts/alerts-list/tree-view/tree-view.component.ts]: Done

**Comment by mmiklavc** [comment]
@merrimanr - can you deconflict this so we can re-test?

**Comment by merrimanr** [comment]
Done

**Comment by james-sirota** [comment]
+1 I am able to verify that when I add the alert in an "open" state to the meta alert the count increments by 1. then i remove it the count decreases by 1

**Pull request closed without merging**